### PR TITLE
[4.2][bugfix] Media manager icons respect the permissions

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue https://github.com/joomla/joomla-cms/issues/36549.

### Summary of Changes

- Pass the persmissions from PHP to JS
- Pass the properties to the icons

### Testing Instructions

Check the issus

### Documentation Changes Required
No, bug

@crystalenka

### CODE REVIEW

**Comment by crystalenka** [comment]
Works on images, there's a minor bug on folders where when you hover the triple-dot icon appears, but when you click it, it disappears again.

screen shot 2022-01-04 at 11 38 43<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/36556">issues.joomla.org/tracker/joomla-cms/36556</a>.</sub>

**Comment by dgrammatiko** [comment]
The problem is the focus management of the drop-down list (the ellipsis button is a dropdown under the skin) because it's kinda hardcoded to at least some of the options. Suddenly the scope of this PR got way bigger...

**Comment by crystalenka** [comment]
Oops. I have a habit of doing that lately. 😬

**Comment by dgrammatiko** [comment]
Oh no, you didn't do anything wrong. Others should be blamed here and I think my name would be also on the list. Anyways this requires quite some refactoring of the buttons and probably I will end up rewriting the whole list view as it's completely inaccessible. That said it's not a 5 mins job...

**Comment by dgrammatiko** [comment]
The whole focus management is completely foobar, that said the code here should do what it was meant to do (hopefully).

**Comment by Quy** [comment]
Folders should not have the pencil icon.

File: administrator/components/com_media/tmpl/media/default.php [hunk 1]
```diff
@@ -54,7 +55,10 @@
 	'providers'           => (array) $this->providers,
 	'currentPath'         => $this->currentPath,
 	'isModal'             => $tmpl === 'component',
-);
+	'canCreate'           => $app->getIdentity()->authorise('core.create', 'com_media'),
+	'canEdit'             => $app->getIdentity()->authorise('core.edit', 'com_media'),
+	'canDelete'           => $app->getIdentity()->authorise('core.delete', 'com_media'),
```
Comment (PhilETaylor) [inline administrator/components/com_media/tmpl/media/default.php:60]: call getIdentity once, and reuse it = 3 times less performance overhead.

Comment (dgrammatiko) [reply administrator/components/com_media/tmpl/media/default.php:60]: Oh nice, will do that later

**Comment by dgrammatiko** [comment]
@laoneo will take over this one

**Pull request closed without merging**